tests/scripts/all.sh

```diff
@@ -6160,63 +6185,87 @@ component_check_test_helpers () {
 }
 
 component_test_psasim() {
-    msg "build library for server"
+    msg "build server library and application"
     scripts/config.py crypto
     helper_crypto_client_build server
 
-    msg "build server"
-    make -C tests/psa-client-server/psasim CFLAGS="$ASAN_CFLAGS" LDFLAGS="$ASAN_CFLAGS" test/psa_partition
-
-    # cleanup() will restore some backed-up files which include $CONFIG_H and
-    # $CRYPTO_CONFIG_H. Built libraries were already copied to psasim at this
-    # point.
-    cleanup
+    cleanup_before_psasim_client
 
     msg "build library for client"
     helper_crypto_client_build client
 
     msg "build psasim to test psa_client"
-    rm -f tests/psa-client-server/psasim/test/psa_client        # In case left behind
-    make -C tests/psa-client-server/psasim CFLAGS="$ASAN_CFLAGS" LDFLAGS="$ASAN_CFLAGS" test/psa_client
+    rm -f $PSASIM_PATH/test/psa_client        # In case left behind
```

Comment:
No, if kept, this should be before `make` is called - specifically, because we were changing what was being built as psa_client, we want to make sure any existing such executable is removed, because `make` uses timestamps to determine what to build, and timestamps on other `.c` files might be older than when the last `psa_client` program was built.

If we're getting rid of all that (haven't reviewed in detail) we probably don't need this `rm` at all

Reply:
`cleanup_before_psasim_client` is meant to be called after the server has been built and before building the client. However please note that `test/psa_client` is _only_ built in `test_psasim` not with standard `test_suite_xxx` and since `test_suite_xxx` is the preferred way of testing for the future, I think we can keep this command only here.
Besides I also think that once we'll have a good coverage of `test_suite_xxx` in the future we can also remove `test_psasim` because that will be superseded by those tests.

Reply:
> in the future we can also remove test_psasim

I think it will be useful to keep this as a smoke test, as if something goes wrong (i.e. we break psasim somehow) it will be very hard to debug once the whole test machinery is involved